### PR TITLE
chore: bump pyathena and related packages

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
Bumping pyathena and adding in the relevant pandas requirements for that package in order to use the schema `pyathena+pandas` which was introduced in 2.0. Typing-extensions was also bumped to 4.x for compatibility with other packages. 

### TESTING INSTRUCTIONS
Creating and running dbs with both rest and pandas drivers for pyathena should now work

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -232,16 +231,14 @@ six==1.16.0
...
-    #   holidays
     #   isodate
     #   jsonschema
     #   polyline
     #   prison
     #   pyrsistent
     #   python-dateutil
-    #   sqlalchemy-utils
     #   wtforms-json
-slack_sdk==3.18.3
```
Comment (pkdotson) [inline requirements/base.txt:244]: Did something change here?

Comment (betodealmeida) [reply requirements/base.txt:244]: The `-` for `_`. ;)

Comment (pkdotson) [reply requirements/base.txt:244]: ah good eye

**Review by pkdotson — APPROVED** [review]

File: requirements/base.txt (hunk 1 above)
Comment (villebro) [reply requirements/base.txt:244]: Good catch. Rant: I always cringe at the fact that `-` and `_` can be used interchangeably for Python deps.. 😡

File: setup.py [hunk 2]
```diff
@@ -120,11 +120,11 @@ def get_git_sha() -> str:
         "sqlparse>=0.4.3, <0.5",
         "tabulate>=0.8.9, <0.9",
         # needed to support Literal (3.8) and TypeGuard (3.10)
-        "typing-extensions>=3.10, <4",
+        "typing-extensions>=4, <5",
```
Comment (villebro) [inline setup.py:123]: Can we add a comment to the description that `typing-extensions` is also bumped in this PR? Also, can you remove the reference to `Literal` on line 122, as we're no longer supporting Python 3.7 or lower (in other words, we currently only need `TypeGuard` from `typing-extensions`)

**Review by villebro — APPROVED** [review]
Nice, bumping always makes me feel warm and fuzzy inside! 😄  LGTM with a few non-blocking comments

File: setup.py (hunk 2 above)
Comment (eschutho) [reply setup.py:122]: sure thing!

**Comment by sadpandajoe** [comment]
🏷️ preset:2022.43

**Pull request merged**